src/Gui/ToolBarManager.cpp

```diff
@@ -1097,9 +1203,47 @@ void Gui::ToolBarManager::setMovable(bool moveable) const
 {
     for (auto& tb : toolBars()) {
         tb->setMovable(moveable);
+        setToolBarMovable(tb);
     }
 }
 
+void ToolBarManager::setToolBarMovable(QToolBar *toolbar) const
+{
+    if (!toolbar)
+        return;
+
+    bool movable = toolbar->isMovable();
+
+    if (auto area = getToolBarArea(toolbar)) {
+        if (auto grip = toolbar->findChild<ToolBarGrip*>()) {
+            if (!movable) {
+                toolbar->removeAction(grip->action());
+                grip->deleteLater();
+            }
+        }
+        else {
+            new ToolBarGrip(toolbar);

```

Comment:
Creating new object without assigning it to anything is a very strange construction. Probably it is fine as it has properly defined parent so Qt will take care of it but nonetheless it is quite odd - please at least put comment over it that it is intentional and how it is supposed to work.